framework/base/Event.php

```diff
@@ -192,5 +203,17 @@ public static function trigger($class, $name, $event = null)
                 }
             }
         } while (($class = get_parent_class($class)) !== false);
```

Comment:
Why don't use `class_parents()` here then?
Both cycles can be unified this way.